woocommerce/woocommerce-ios
# [Analytics Hub] Display error messages after an error retrieving order stats

Part of #8213

## Description

This PR adds support for displaying an error message on each card in the Analytics Hub when there is an error retrieving order stats. Before, we were just showing an empty state with no clear error message.

**Note:** The error message is displayed below the placeholder data on each card. Right now, the delta tag placeholders stick out with a red background, but that will be updated to gray in #8243.

## Changes

* Each analytics card now has a `showSyncError` property to determine if the error should be shown, and a unique error message for that card.
* In `AnalyticsHubViewModel`, if either the current or previous order stats are `nil` when a card's view model is being created, that card's `showSyncError` property will be set to `true`.
* Updates tests to confirm that each card state works as expected (showing data when it's available, redacted while the network request is made, and showing an error when the network returns an error).

## Testing

1. Disable your network connection (to cause a sync error).
2. Launch the app.
3. Tap the "See more" button on the My Store dashboard.
4. Notice the analytics cards display placeholder data and a clear error message.

## Screenshots

Before|After
-|-
![Empty state](https://user-images.githubusercontent.com/8658164/205034123-7210ad21-dc96-4744-86d7-037c330fff32.png)|![Sync errors](https://user-images.githubusercontent.com/8658164/205099162-f549c8f2-35be-44ca-8b72-45356651f665.png)

https://user-images.githubusercontent.com/8658164/205099239-29968626-946c-4d73-8c5f-2a28daa12559.mp4

## Submitter Checklist

Update release notes:
